src/tabslide.ts

```diff
@@ -39,38 +42,44 @@ export class TabSlide extends TabTie {
     this.setAttribute('type', 'TabSlide');
 
     if (!direction) {
-      const first_fret = notes.first_note.getPositions()[0].fret;
-      const last_fret = notes.last_note.getPositions()[0].fret;
+      const first_fret = ((notes.first_note as unknown) as TabNote).getPositions()[0].fret;
+      const last_fret = ((notes.last_note as unknown) as TabNote).getPositions()[0].fret;
 
       direction = parseInt(first_fret, 10) > parseInt(last_fret, 10) ? TabSlide.SLIDE_DOWN : TabSlide.SLIDE_UP;
     }
 
-    this.slide_direction = direction;
+    this.direction = direction;
     this.render_options.cp1 = 11;
     this.render_options.cp2 = 14;
     this.render_options.y_shift = 0.5;
 
-    this.setFont({ font: 'Times', size: 10, style: 'bold italic' });
+    this.setFont({ family: 'Times', size: 10, weight: 'bold italic' });
     this.setNotes(notes);
   }
 
-  renderTie(params) {
+  renderTie(params: {
+    direction: number;
+    first_x_px: number;
+    last_x_px: number;
+    last_ys: number[];
+    first_ys: number[];
+  }): void {
     if (params.first_ys.length === 0 || params.last_ys.length === 0) {
       throw new Vex.RERR('BadArguments', 'No Y-values to render');
     }
 
-    const ctx = this.context;
+    const ctx = this.checkContext();
     const first_x_px = params.first_x_px;
     const first_ys = params.first_ys;
     const last_x_px = params.last_x_px;
 
-    const direction = this.slide_direction;
+    const direction = params.direction;
```

Comment:
I think this should be `this.direction`

Reply:
Done. The result is also fine. However, this means that params.direction is not used, is this ok?

Reply:
I'm confused about this... we have both `this.slide_direction` and `params.direction`, so it seems like we can't just ignore `this.direction`. How about `const direction = params.direction ?? this.slide_direction`? (And make `params.direction` optional?)

Reply:
Okay, actually, I dug into the code and see what's going on now -- `slide_direction` is different from `direction`, and this class does not implement `draw`. So your original `this.slide_direction` was correct. Sorry about that, can you revert this line please?

Reply:
done :)